### PR TITLE
Disable static restore in more locations

### PR DESCRIPTION
**Pull request by dibarbet**
took a quick stab at fixing this

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -206,6 +206,7 @@ stages:
           include:
           - src/Compilers/*
           - src/Dependencies/*
+          - eng/*
```
Comment (dibarbet) [inline azure-pipelines.yml:209]: seems like we should run compiler CI when build scripts change

Comment (Youssef1313) [reply azure-pipelines.yml:209]: Since you are touching this, could you also fix https://github.com/dotnet/roslyn/issues/65602 in this PR?

Comment (dibarbet) [reply azure-pipelines.yml:209]: Added the compiler generator tools path - let me know if there were other directories you were thinking of

**Review by jaredpar — APPROVED** [review]

**Review by Cosifne — APPROVED** [review]

**Comment by RikkiGibson** [comment]
Is there a writeup for why disabling static restore fixes correctness jobs?

**Comment by dibarbet** [comment]
> Is there a writeup for why disabling static restore fixes correctness jobs?

It isn't just correctness - it technically affects all the builds with static restore (see the nuget restore issue).  However the reason it hit correctness builds first (well technically it hit the scouting queue integration tests first) is that they are specifically requiring an image with VS preview on it (17.5) which has a version of nuget with the issue.

**Comment by dibarbet** [comment]
@RikkiGibson or @jaredpar - any idea on what might be happening with the rebuild check?  It passed on the previous commit (and I only changed the folders qualifying as a compiler change in the yaml which I don't believe should impact correctness).

I already re-ran once and hit the same error

**Comment by RikkiGibson** [comment]
It looks like the bootstrap build is failing in that job. Why only in Rebuild? I don't think any rebuild-specific stuff is happening at that point.

**Comment by jaredpar** [comment]
Merging to help unblock PRs.

**Pull request merged**